examples/scanvi/data.py

```diff
@@ -0,0 +1,159 @@
+            _slice = slices[batch_order[i]]
+            if _slice[1]:
+                # labeled
+                yield self.data_x[_slice[0]], \
+                      nn.functional.one_hot(self.data_y[_slice[0]], num_classes=self.num_classes)
+            else:
+                # unlabeled
+                yield self.data_x[_slice[0]], None
+
+
+def _get_score(normalized_adata, gene_set):
+    """
+    Returns the score per cell given a dictionary of + and - genes
+    """
+    score = np.zeros(normalized_adata.n_obs)
+    for gene in gene_set['positive']:
+        expression = np.array(normalized_adata[:, gene].X)
+        score += expression.flatten()
+    for gene in gene_set['negative']:
+        expression = np.array(normalized_adata[:, gene].X)
+        score -= expression.flatten()
+    return score
+
+
+def _get_cell_mask(normalized_adata, gene_set):
+    """
+    Calculates the score per cell for a list of genes, then returns a mask for
+    the cells with the highest 50 scores.
+    """
+    score = _get_score(normalized_adata, gene_set)
+    cell_idx = score.argsort()[-50:]
+    mask = np.zeros(normalized_adata.n_obs)
+    mask[cell_idx] = 1
+    return mask.astype(bool)
+
+
+def get_data_loader(batch_size=100, cuda=False):
+    """
+    Does the necessary preprocessing and returns a BatchDataLoader for the PBMC dataset.
+    """
+    adata = scvi.data.purified_pbmc_dataset(subset_datasets=["regulatory_t", "naive_t",
+                                                             "memory_t", "naive_cytotoxic"])
+
+    gene_subset = ["CD4", "FOXP3", "TNFRSF18", "IL2RA", "CTLA4", "CD44", "TCF7",
+                   "CD8B", "CCR7", "CD69", "PTPRC", "S100A4"]
+
+    normalized = adata.copy()
+    sc.pp.normalize_total(normalized, target_sum=1e4)
+    sc.pp.log1p(normalized)
+
+    normalized = normalized[:, gene_subset].copy()
+    sc.pp.scale(normalized)
+
+    # hand curated list of genes for identifying ground truth
+    cd4_reg_geneset = {"positive": ["TNFRSF18", "CTLA4", "FOXP3", "IL2RA"], "negative": ["S100A4", "PTPRC", "CD8B"]}
+    cd8_naive_geneset = {"positive": ["CD8B", "CCR7"], "negative": ["CD4"]}
+    cd4_naive_geneset = {"positive": ["CCR7", "CD4"], "negative": ["S100A4", "PTPRC", "FOXP3", "IL2RA", "CD69"]}
+    cd4_mem_geneset = {"positive": ["S100A4"], "negative": ["IL2RA", "FOXP3", "TNFRSF18", "CCR7"]}
+
+    cd4_reg_mask = _get_cell_mask(normalized, cd4_reg_geneset)
+    cd8_naive_mask = _get_cell_mask(normalized, cd8_naive_geneset)
+    cd4_naive_mask = _get_cell_mask(normalized, cd4_naive_geneset)
+    cd4_mem_mask = _get_cell_mask(normalized, cd4_mem_geneset)
+
+    seed_labels = -np.ones(cd4_mem_mask.shape[0])
+    seed_labels[cd8_naive_mask] = 0  # "CD8 Naive T cell"
+    seed_labels[cd4_naive_mask] = 1  # "CD4 Naive T cell"
+    seed_labels[cd4_mem_mask] = 2    # "CD4 Memory T cell"
+    seed_labels[cd4_reg_mask] = 3    # "CD4 Regulatory T cell"
+
+    # filter out non-variable genes
```

Comment:
@romain-lopez is this filtering logic reasonable in this context? i'm a big confused about how these scanpy objects work. in particular can i now use `adata` to make UMAP plots using scanpy?